roles/gpu/tasks/gpu.yml

```diff
@@ -0,0 +1,67 @@
+---
+- name: Install yum requirements for gpu driver installation
+  ansible.builtin.yum:
+    state: 'installed'
+    update_cache: true
+    name:
+      - tar
+      - bzip2
+      - make
+      - automake
+      - gcc
+      - gcc-c++
+      - pciutils
+      - elfutils-libelf-devel
+      - libglvnd-devel
+      - bind-utils
+      - wget
+  become: true
+
+# ansible_kernel variable is not working, as after reboot, still holds old kernel
+- name: Get current kernel version
+  ansible.builtin.command: '/usr/bin/uname -r'
+  register: uname_output
+  failed_when: uname_output.rc != 0
+  when: true
+  become: true
+
+- name: Set kernel version fact
+  ansible.builtin.set_fact:
+    kernel_version: "{{ uname_output.stdout }}"
+
+- name: Install kernel developement package matching running kernel version
+  ansible.builtin.yum:
+    name: 'kernel-devel-{{ kernel_version }}'
+  register: yum_result
+  failed_when: yum_result.rc != 0
+  when: true
+  become: true
+
+- name: Download a driver installation file from NVidia
+  ansible.builtin.get_url:
+    url: '{{ gpu_url_directory }}/{{ gpu_runfile }}'
+    dest: '/root/{{ gpu_runfile }}'
+    mode: '0700'
+  become: true
+
+- name: "Check if driver downloaded"
+  ansible.builtin.stat:
+    path: '/root/{{ gpu_runfile }}'
+  when: true
+  register: driver_downloaded
+  become: true
+
+- name: Install driver from .run file
+  ansible.builtin.command: '/root/{{ gpu_runfile }} --silent --driver'
+  register: install_result
+  failed_when: install_result.rc != 0
+  when: driver_downloaded.stat.exists
```

Comment:
Is this "when" necessary? If the driver was not successfully downloaded (either now or during a previous play), this task should fail. Instead it will now get skipped and nothing will be installed and there will not be an error either.

Reply:
Ansible-lint expects the steps with `ansible.builtin.command` to have `when` defined. I guess this is the reasoning on why I add it here...
If the driver download would fail, then this (as the step before installing) would fail first.
I don't know if it make sense to change it into `when: true`?

Reply:
Ansible-lint wants you to make sure the task is idempotent, which requires "something" to determine if the command should be executed or skipped, because it has already been used successfully before. Adding `when` is one way to make it idempotent, but condition you added for the `when` does not make sense. If this `gpu.yml` is only included when a GPU was detected and drivers are missing, then that ensures idempotency and you can just use `when: true` here.

Reply:
fixed